beacon-chain/sync/initial-sync/blocks_fetcher.go

```diff
@@ -1089,15 +1201,43 @@ func processDataColumn(
 // - `missingColumnsByRoot` by removing the fetched data columns.
 func (f *blocksFetcher) fetchDataColumnFromPeer(
 	ctx context.Context,
+	mu *sync.RWMutex,
```

Comment:
Usually, we use mutexes as struct field because mutexes are protecting struct fields as well.
Here, `mu` is only protecting concurrent access to `bwbs` and `missingColumnsByRoot`, which are not struct fields, but variables defined inside a function. That's why the corresponding mutex `mu` does not need to be defined as a struct field, and so the mutex is passed as an argument.

Reply:
is it possible to pass in the shared resource rather than just the mutex ? Basically have all these items enclosed in a single logical object

Reply:
Fixed in 4b4ba601ee38d73b7ee7366fbce0ba275eabb8c2.